x/epochstorage/keeper/stake_storage.go

```diff
@@ -225,7 +225,7 @@ func (k Keeper) ModifyStakeEntryCurrentFromStorage(ctx sdk.Context, stakeStorage
 	// remove the given index, then store the new entry in the sorted list at the right place
 	entries := []types.StakeEntry{}
 	for i, entry := range stakeStorage.StakeEntries {
-		if entry.Address == stakeEntry.Address {
+		if entry.Operator == stakeEntry.Operator || entry.Vault == stakeEntry.Vault {
```

Comment:
why not &&

Reply:
see comment above

Reply:
here they must match, if not, we have a problem

Reply:
done. see https://github.com/lavanet/lava/commit/1666a41c95c727bf777d20ac366efef3b31ff54e